src/mame/drivers/didact.cpp

```diff
@@ -60,31 +60,43 @@
 // Features
 #include "imagedev/cassette.h"
 #include "bus/rs232/rs232.h"
-
 #include "screen.h"
 
 // Generated artwork includes
 #include "mp68a.lh"
 #include "md6802.lh"
 
-#define VERBOSE 0
+//**************************************************************************
+//  MACROS / CONSTANTS
+//**************************************************************************
 
-#define LOGPRINT(x) do { if (VERBOSE) logerror x; } while (0)
-#define LOG(x) LOGPRINT(x)
-#define LOGSCAN(x) {}
-#define LOGSER(x) {}
-#define LOGSCREEN(x) {}
-#define RLOG(x) {}
-#define LOGCS(x) {}
+//#define LOG_GENERAL (1U <<  0) // Already defined in logmacro.h
+#define LOG_SETUP   (1U <<  1)
+#define LOG_SCAN    (1U <<  2)
+#define LOG_SERIAL  (1U <<  3)
+#define LOG_SCREEN  (1U <<  4)
+#define LOG_READ    (1U <<  5)
+#define LOG_CS      (1U <<  6)
 
-#if VERBOSE >= 2
-#define logerror printf
-#endif
+//#define VERBOSE  (LOG_GENERAL|LOG_SETUP|LOG_SCREEN)
+//#define LOG_OUTPUT_FUNC printf
+
+#include "logmacro.h"
+
+//#define LOG(...)        LOGMASKED(LOG_GENERAL, __VA_ARGS__) // Already defined in logmacro.h
+#define LOGSETUP(...)  LOGMASKED(LOG_SETUP, __VA_ARGS__)
+#define LOGSCAN(...)   LOGMASKED(LOG_SCAN, __VA_ARGS__)
+#define LOGSER(...)    LOGMASKED(LOG_SERIAL, __VA_ARGS__)
+#define LOGSCREEN(...) LOGMASKED(LOG_SCREEN, __VA_ARGS__)
+#define LOGR(...)      LOGMASKED(LOG_READ, __VA_ARGS__)
+#define LOGCS(...)     LOGMASKED(LOG_CS, __VA_ARGS__)
 
 #ifdef _MSC_VER
 #define FUNCNAME __func__
+#define LLFORMAT "%I64d"
```

Comment:
This won't work properly.  The required format specifier doesn't depend on whether you're using MSVC or not, it's more complex than that.  The msys64 runtime also uses `I64` as the size qualifier for `long long` integers for example.  But that aside, `logerror` and friends don't need size qualifiers for integer formatting, so there's no need for it.  Please stop adding this to things that don't absolutely need it.